apollo-router/tests/integration/file_upload.rs

```diff
@@ -22,6 +22,125 @@ macro_rules! make_handler {
     }
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn it_uploads_file_to_subgraph() -> Result<(), BoxError> {
```

Comment:
My best effort is writing tests that test stuff that is sent to the subgraph.
I chose not to use the existing test harness and made this test as simple as possible.
My initial plan was to update the existing test to snapshot stuff sent to the subgraph, but I failed to finish it in the time allocated for this fix, so I just wrote a super basic test that tests the subgraph request in the most basic scenario.